### PR TITLE
use quay.io qm - kvm

### PR DESCRIPTION
**Pull request by dougsland**

### CODE REVIEW

**Comment by dougsland** [comment]
I was working/discussing with @Yarboa about this one. 
cc @aesteve-rh

File: rpm/qm.spec [hunk 1]
```diff
@@ -223,6 +223,9 @@ install -d %{buildroot}%{_sysconfdir}/qm/containers/containers.conf.d
 # START - qm dropin sub-package - mount kvm            #
 ########################################################
 %if %{enable_qm_mount_bind_kvm}
+    # first step - add drop-in file in /etc/containers/containers.d.conf/qm_dropin_mount_bind_snd.conf
```
Comment (Yarboa) [inline rpm/qm.spec:226]: If gate is passing, it means that this part is failing the CI
I do not think it is important
That one below is enough 
```
%{_builddir}/qm-%{version}/etc/qm/containers/containers.conf.d/qm_dropin_mount_bind_kvm.conf %{buildroot}%
```

**Review by Yarboa — COMMENTED** [review]
I am not sure I am following the use case, 
Is it containerized vm inside QM., correct? 
So I think it obsoletes the previous work done for kvm, isnt it? 
I think README should be changed, also, do you want to maintain both cases?

File: rpm/qm.spec (hunk 1 above)
Comment (dougsland) [reply rpm/qm.spec:226]: @Yarboa If you say, I trust. I was just cascading /dev/kvm from host -> QM -> nested containers.

**Comment by dougsland** [comment]
> I am not sure I am following the use case, Is it containerized vm inside QM., correct? So I think it obsoletes the previous
> work done for kvm, isnt it? I think README should be changed, also, do you want to maintain both cases?

You are correct, as soon it works we need to update README.md.

File: rpm/qm.spec (hunk 1 above)
Comment (Yarboa) [reply rpm/qm.spec:226]: It has improved the status of that #660 
I am testing manually but i have selinux issues due to disk partition extension

File: subsystems/kvm/ContainerFile [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+                   guestfs-tools \
+                   wget \
+                   vim -y \
+    && dnf clean all && rm -rf /var/cache/dnf
+
+RUN wget -O /var/lib/libvirt/images/Fedora-Cloud-Base-Generic.qcow2 https://cofractal-ewr.mm.fcix.net/fedora/linux/releases/41/Cloud/x86_64/images/Fedora-Cloud-Base-Generic-41-1.4.x86_64.qcow2
+
+# Set fedora user pass as fedora
+RUN export LIBGUESTFS_BACKEND=direct && \
+	virt-customize -a /var/lib/libvirt/images/Fedora-Cloud-Base-Generic.qcow2 --password fedora:password:$PASSWORD_FEDORA_USER
```
Comment (Yarboa) [inline subsystems/kvm/ContainerFile:26]: Please do no forget to remove cloud init
and allow root to login 

```
virt-customize --uninstall cloud-init --root-password password:Fed123 --edit '/etc/ssh/sshd_config: s/#PermitRootLogin.*/PermitRootLogin yes/' -a <PATH_TO_IMAGE_DIR><FEDORA_IMAGE>
```

File: subsystems/kvm/ContainerFile [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+                   libvirt-daemon \
+                   libvirt-daemon-qemu \
+                   libvirt-daemon-kvm \
+                   libvirt-daemon-config-network \
+                   guestfs-tools \
+                   wget \
+                   vim -y \
+    && dnf clean all && rm -rf /var/cache/dnf
+
+RUN wget -O /var/lib/libvirt/images/Fedora-Cloud-Base-Generic.qcow2 https://cofractal-ewr.mm.fcix.net/fedora/linux/releases/41/Cloud/x86_64/images/Fedora-Cloud-Base-Generic-41-1.4.x86_64.qcow2
```
Comment (Yarboa) [inline subsystems/kvm/ContainerFile:22]: I would also use ENV_VAR for changing fedora mirror
Default should be 
```
https://download.fedoraproject.org/pub/fedora/linux/releases/
```

https://download.fedoraproject.org/pub/fedora/linux/releases/41/Cloud/x86_64/images/Fedora-Cloud-Base-Generic-41-1.4.x86_64.qcow2

File: subsystems/kvm/ContainerFile [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+
+# Set fedora user pass as fedora
+RUN export LIBGUESTFS_BACKEND=direct && \
+	virt-customize -a /var/lib/libvirt/images/Fedora-Cloud-Base-Generic.qcow2 --password fedora:password:$PASSWORD_FEDORA_USER
+
+# Permission to qemu user/group
+RUN chown qemu:qemu /var/lib/libvirt/images/Fedora-Cloud-Base-Generic.qcow2
+
+# Set systemd as the init system for the container
+ENTRYPOINT ["/usr/sbin/init"]
```
Comment (Yarboa) [inline subsystems/kvm/ContainerFile:32]: I think it should run qemu isnt it? 
We would like kvm to on start

File: subsystems/kvm/ContainerFile (hunk 3 above)
Comment (dougsland) [reply subsystems/kvm/ContainerFile:22]: Hey @Yarboa Are you going to update this as well ?

File: subsystems/kvm/ContainerFile (hunk 2 above)
Comment (Yarboa) [reply subsystems/kvm/ContainerFile:26]: Prefer all be tested before merged

File: subsystems/kvm/ContainerFile (hunk 3 above)
Comment (Yarboa) [reply subsystems/kvm/ContainerFile:22]: Prefer all be tested before merged

Comment (dougsland) [reply subsystems/kvm/ContainerFile:22]: I tested.

**Comment by Yarboa** [comment]
Please review #670 
Rebase and lets merge this one.

**Comment by Yarboa** [comment]
@dougsland 
I am removing the image download and guestfs-tools
Once image created, 
We should download image, virt-cust and podman cp image qcow2 image

Current image goes up to 3.5GB it is too big

**Comment by dougsland** [comment]
> @dougsland I am removing the image download and guestfs-tools Once image created, We should download image, virt-cust and podman cp image qcow2 image
> 
> Current image goes up to 3.5GB it is too big

I trust on you, just update the PR or create a new one with your ideas.

File: subsystems/kvm/ContainerFile [hunk 5]
```diff
@@ -3,30 +3,16 @@
...
-                   libvirt-daemon-kvm \
-                   libvirt-daemon-config-network \
-                   guestfs-tools \
-                   wget \
-                   vim -y \
-    && dnf clean all && rm -rf /var/cache/dnf
+RUN dnf install qemu-system-$(arch) -y \
+                 && dnf clean all && rm -rf /var/cache/dnf
 
-RUN wget -O /var/lib/libvirt/images/Fedora-Cloud-Base-Generic.qcow2 https://cofractal-ewr.mm.fcix.net/fedora/linux/releases/41/Cloud/x86_64/images/Fedora-Cloud-Base-Generic-41-1.4.x86_64.qcow2
```
Comment (dougsland) [inline subsystems/kvm/ContainerFile]: We don't need to download this file anymore?

Comment (dougsland) [reply subsystems/kvm/ContainerFile]: found

**Pull request merged**